src/main/java/kr/hhplus/be/server/infrastructure/external/naver/NaverApiClientImpl.java

```diff
@@ -0,0 +1,129 @@
+package kr.hhplus.be.server.infrastructure.external.naver;
+
+
+import kr.hhplus.be.server.infrastructure.external.api.SearchApiClient;
+import kr.hhplus.be.server.infrastructure.external.dto.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class NaverApiClientImpl implements SearchApiClient {
+
+    private static final String NAVER_LOCAL_SEARCH_URL = "https://openapi.naver.com/v1/search/local.json";
+
+    @Value("${naver.client.id}")
+    private String clientId;
+
+    @Value("${naver.client.secret}")
+    private String clientSecret;
+
+    private final RestTemplate restTemplate;
+    private final ObjectMapper objectMapper;
+
+    public NaverApiClientImpl(RestTemplate restTemplate, ObjectMapper objectMapper) {
```

Comment:
```suggestion
public NaverApiClientImpl(
    @Value("${naver.client.id}") String clientId,
    @Value("${naver.client.secret}") String clientSecret,
    RestTemplate restTemplate,
    ObjectMapper objectMapper
)
}
```

생성자 주입이 스프링에서 권장되는 방식이긴 합니다 ㅎㅎ